### PR TITLE
PPTP-1516 - Move get and update end points to registration service

### PR DESCRIPTION
**Pull request by kiranprakashdigital**
### Description of Work carried through

Brief description of what/why/how.

#### Check list 
 - [ ] `./precheck` was executed (Integration/Component/Unit tests)
 - [ ] `sbt scalafmt test:scalafmt` was executed
 - [ ] Required Environment Config has been amended/added
 - [ ] Links to dependencies have been included (BE/FE work)
 - [ ] User Acceptance Tests (UAT) were run locally and have passed.

### CODE REVIEW

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/SubscriptionsConnectorISpec.scala [hunk 1]
```diff
@@ -434,6 +436,206 @@ class SubscriptionsConnectorISpec
...
+                .withStatus(Status.CONFLICT)
+                .withBody(Json.obj("xxx" -> "xxx").toString)
+            )
+        )
+
+        intercept[UpstreamErrorResponse] {
+          await(connector.updateSubscription(pptReference, ukLimitedCompanySubscription))
+        }.statusCode mustBe Status.INTERNAL_SERVER_ERROR
+      }
+    }
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/SubscriptionsConnectorISpec.scala:638]: Each of these tests is very similar. Could we refactor as a single table driven test in order to reduce code duplication?

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/SubscriptionsConnectorISpec.scala [hunk 2]
```diff
@@ -434,6 +436,206 @@ class SubscriptionsConnectorISpec
...
+              aResponse()
+                .withStatus(Status.OK)
+                .withBody(
+                  Json.obj("pptReferenceNumber" -> pptReference,
+                           "processingDate"     -> subscriptionProcessingDate,
+                           "formBundleNumber"   -> formBundleNumber
+                  ).toString
+                )
+            )
+        )
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/SubscriptionsConnectorISpec.scala:457]: I think we should extract this stubbing into a private method as we've done for a failed update.

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/SubscriptionControllerSpec.scala [hunk 3]
```diff
@@ -205,7 +206,181 @@ class SubscriptionControllerSpec
...
+
+  "Get subscription" should {
+    "return expected details" in {
+      withAuthorizedUser()
+      mockGetSubscription(ukLimitedCompanySubscription)
+
+      val result: Future[Result] = route(app, subscriptionResponse_HttpGet).get
+
+      status(result) must be(OK)
+      //      contentAsJson(result) mustBe toJson(subscriptionStatusResponse)
```
Comment (RyanKellyHMRC) [inline test/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/SubscriptionControllerSpec.scala:226]: Should this assertion have been reinstated?

**Review by RyanKellyHMRC — CHANGES_REQUESTED** [review]
Looks good. Just a couple of small changes requested.

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/SubscriptionControllerSpec.scala (hunk 3 above)
Comment (kiranprakashdigital) [reply test/uk/gov/hmrc/plasticpackagingtaxregistration/controllers/SubscriptionControllerSpec.scala]: Done

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/SubscriptionsConnectorISpec.scala (hunk 2 above)
Comment (kiranprakashdigital) [reply test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/SubscriptionsConnectorISpec.scala:457]: Done

File: test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/SubscriptionsConnectorISpec.scala (hunk 1 above)
Comment (kiranprakashdigital) [reply test/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/SubscriptionsConnectorISpec.scala:373]: Done

**Review by RyanKellyHMRC — APPROVED** [review]
Nice one Kiran. Loads of duplicated code removed.

**Pull request merged**